pkg/cmd/server/start/start_node.go

```diff
@@ -426,28 +421,77 @@ func execKubelet(kubeletArgs []string) (bool, error) {
 			break
 		}
 	}
+	// execve the child process, replacing this process
 	glog.V(3).Infof("Exec %s %s", kubeletPath, strings.Join(args, " "))
-	return false, syscall.Exec(kubeletPath, args, os.Environ())
+	return syscall.Exec(kubeletPath, args, os.Environ())
 }
 
+// safeArgRegexp matches only characters that are known safe. DO NOT add to this list
+// without fully considering whether that new character can be used to break shell escaping
+// rules.
+var safeArgRegexp = regexp.MustCompile(`^[\da-zA-Z\-=_\.,/\:]+$`)
+
+// shellEscapeArg quotes an argument if it contains characters that my cause a shell
+// interpreter to split the single argument into multiple.
+func shellEscapeArg(s string) string {
+	if safeArgRegexp.MatchString(s) {
+		return s
+	}
+	return strconv.Quote(s)
+}
+
+// argsMapToArgs writes a sorted list of arguments for flags. It does not escape the
+// flag values.
+func argsMapToArgs(argsMap map[string][]string) []string {
+	var keys []string
+	for key := range argsMap {
+		keys = append(keys, key)
+	}
+	sort.Strings(keys)
+
+	var args []string
+	for _, key := range keys {
+		for _, token := range argsMap[key] {
+			args = append(args, fmt.Sprintf("--%s=%v", key, token))
+		}
+	}
+	return args
+}
+
+// WriteKubeletFlags writes the correct set of flags to start a Kubelet from the provided node config to
+// stdout, instead of launching anything.
+func WriteKubeletFlags(nodeConfig configapi.NodeConfig) error {
+	kubeletFlagsAsMap, err := nodeoptions.ComputeKubeletFlagsAsMap(nodeConfig.KubeletArguments, nodeConfig)
```

Comment:
I meant: couldn't this just return the string slice of args?  It's the first thing you do at both call sites

Reply:
Oh.